.github/workflows/ci-actions-incremental.yml

```diff
@@ -498,6 +498,13 @@ jobs:
       - name: Extract Maven Repo
         shell: bash
         run: tar -xzf maven-repo.tgz -C ~
+      - name: Cache Gradle Repository
+        id: cache-gradle
+        uses: actions/cache@v2
+        with:
+          path: ~/.gradle/caches
+          # refresh cache every month to avoid unlimited growth
+          key: q2gradle-${{ steps.get-date.outputs.date }}
```

Comment:
Just an idea but maybe we could try to use the cache feature integrated in `setup-java@v3`? More details here: https://github.com/actions/setup-java#caching-packages-dependencies

Reply:
I updated the branch using the `setup-java@v3`